android/src/main/java/nordpol/TagDispatcher.java

```diff
@@ -79,9 +87,10 @@ public boolean disableExclusiveNfc() {
             } else {
                 disableForegroundDispatch(adapter);
             }
-            return true;
+            return NfcStatus.AVAILABLE;
         }
-        return false;
+        if (handleUnavailableNfc) toast("NFC is not available on this device");
```

Comment:
I'd argue that it is not overkill. The parameter says that the library should handle unavailable NFC. If we don't do this the implementer still has to get the adapter and all to even have basic support. This way we facilitate a pretty good UX right from the get-go.